docs/introduction.asciidoc

```diff
@@ -5,29 +5,31 @@ https://www.elastic.co/guide/en/apm/agent/ruby/current/introduction.html[elastic
 endif::[]
 
 [[introduction]]
-
 == Introduction
 
-Welcome to the APM Ruby Agent documentation.
-
-The Elastic APM Ruby Agent sends performance metrics and error logs to an
-Elastic APM Server.
+The Elastic APM Ruby Agent sends performance metrics and error logs to the APM Server.
 It has built-in support for <<getting-started-rails,Ruby on Rails>> and other
 <<getting-started-rack,Rack-compatible>> applications.
+It also offers an API which allows you to instrument any application.
+
+[float]
+[[how-it-works]]
+=== How does the Agent work?
+
+The agent auto-instruments <<supported-technologies,supported technologies>> and records interesting events,
+like HTTP requests and database queries. To do this, it {NEED CONTENT HERE}.
+This means that for the supported technologies, there are no code changes required.
+
+The Agent automatically {NEED CONTENT HERE} to measure their duration and metadata (like the DB statement),
+as well as HTTP related information (like the URL, parameters, and headers).
+
+These events, called Transactions and Spans, are sent to the APM Server.
+The APM Server converts them to a format suitable for Elasticsearch, and sends them to an Elasticsearch cluster.
+You can then use the APM app in Kibana to gain insight into latency issues and error culprits within your application.
```

Comment:
@mikker / @estolfo -- Part of this PR is adding a section on how the Agent does instrumentation. This doesn't have to be super detailed, just an overview is fine. I've included a template here with some `{NEED CONTENT HERE}` markers. We don't have to follow this template if you have a better way to explain things. Anyway, I'll need some feedback from you to help fill in the blanks here.